### PR TITLE
HPCC-20858 Clean up of HPCCFile interface

### PR DESCRIPTION
**Pull request by rpastrana**
- Removes confusing constructors
- Provides Connection based constructors
- Adds setter methods
- Restructures and cleans up tests mains

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

### CODE REVIEW

**Comment by rpastrana** [comment]
@jpmcmu please review

File: DataAccess/src/main/java/org/hpccsystems/spark/HpccFile.java [hunk 1]
```diff
@@ -42,235 +44,216 @@
...
 
+  private static final Logger       log          = Logger.getLogger(HpccFile.class.getName());
+
   private DataPartition[] dataParts;
   private RecordDef recordDefinition;
   private boolean isIndex;
   static private final int DEFAULT_ACCESS_EXPIRY_SECONDS = 120;
   private int fileAccessExpirySecs = DEFAULT_ACCESS_EXPIRY_SECONDS;
 
+  private Connection espConnInfo;
```
Comment (jpmcmu) [inline DataAccess/src/main/java/org/hpccsystems/spark/HpccFile.java]: Can we make the Connection info transient so it won't be serialized? We will need to test this on a multi-node cluster but this should only be needed on the driver at the moment, so it should be fine if it isn't replicated to the slave nodes.

File: DataAccess/src/main/java/org/hpccsystems/spark/HpccFile.java [hunk 2]
```diff
@@ -42,235 +44,216 @@
...
   {
-  this(fileName, protocol, host, port, user, pword, targetColumnList, filter, remap_info, maxParts, "");
-  this.fileAccessExpirySecs = fileAccessExpirySecs;
+	return clusterRemapInfo;
   }
+
+  /**
+  * @param remapinfo
+  */
+  public void setclusterRemapInfo(RemapInfo remapinfo)
```
[the remark this answers is not in the document]
Comment (rpastrana) [reply DataAccess/src/main/java/org/hpccsystems/spark/HpccFile.java]: yes! Eclipse failed me when it generated these for me...

**Comment by rpastrana** [comment]
@jpmcmu please review

**Comment by jpmcmu** [comment]
@rpastrana looks good

**Comment by rpastrana** [comment]
@richardkchapman please merge

File: DataAccess/src/main/java/org/hpccsystems/spark/HpccFile.java [hunk 3]
```diff
@@ -42,235 +44,216 @@
...
+  /**
+  * @param filterexpression
+  */
+  public void setFilter(String filterexpression)
+  {
+	this.filter = new FileFilter(filterexpression);
+  }
+
+  /**
+  * @return
```
Comment (richardkchapman) [inline DataAccess/src/main/java/org/hpccsystems/spark/HpccFile.java]: Not a showstopper but some of these javadoc comments look incomplete?

**Comment by rpastrana** [comment]
@richardkchapman squashed

**Pull request merged**